icegatherer.go

```diff
@@ -11,6 +11,17 @@ import (
 	"github.com/pion/logging"
 )
 
+// ICEGatherOptions provides options relating to the gathering of ICE candidates.
+type ICEGatherOptions struct {
```

Comment:
This comes from the RFC. If possible I would avoid diverging!

https://draft.ortc.org/#dom-rtcicegatheroptions

Reply:
sure, do you want me to add an extra config object instead?

Reply:
Can you pass the options to `NewGatherer` [here](https://github.com/pion/webrtc/blob/master/internal/ice/gatherer.go#L41)

Reply:
It is ugly having some many arguments to that function, but I think it is important we don't diverge from the standard (I think)

I would love to figure out how we can push these things into the WebRTC standard.

Reply:
cool!, I changed it, however regarding the bigger picture I think there is an argument to be made regarding how adding additional options is not "diverging" from the standard. Specially in this case, because this new options (as the standard specifies) are up to the implementation to decide.